### PR TITLE
Sebastian Sloan

### PR DESCRIPTION
**Pull request by sasloan**
I got to user story 4 but I found it to be a little bit confusing. I think you wanted us to create a form in the movies show page where a user could type in the name of an actor and if that actor existed in the db then it would add that actor too the movie? If that is right then I got hung up on my controller adding the wrong actor. I could not seem to get it to compare what was submitted in the form to the database in order to find it. I feel like I was close but my controller method needs to be tweeked a bit to actually find what was submitted in the database.

### CODE REVIEW

File: app/controllers/actors_controller.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class ActorsController < ApplicationController
+
+	def show
+		@actor = Actor.find(params[:actor_id])
+	end
+end
```
[the remark this answers is not in the document]
Comment (sasloan) [reply app/controllers/actors_controller.rb]: Not sure why it looks this way on github? my Atom shows it as spaced correctly?

File: app/controllers/add_actors_controller.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class AddActorsController < ApplicationController
```
[the remark this answers is not in the document]
Comment (sasloan) [reply app/controllers/add_actors_controller.rb]: I see so when we are making a new controller in order for it to stay RESTful I would re name it to movie_actors_controller.rb and rename the class too MovieActorsController got it I see

File: app/views/movies/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+<h2><%= @movie.name %></h2>
+	<p>Creation Year: <%= @movie.creation_year %></p>
+	<p>Genre: <%= @movie.genre %></p>
+
+<h4> Actors </h4>
+
+<p><%= @movie.actors.list_alphabetically %></p>
+
+<p>Average Age of Actors: <%= @movie.actors.average_actor_age.round(0) %></p>
```
[the remark this answers is not in the document]
Comment (sasloan) [reply app/views/movies/show.html.erb]: So I don't understand. I am calling this on my movies table not my movie_actors table. how else would I get the average age of actors if I didn't put in @movie.actors? isn't that restfull? or did you want me to put the "average_actor_age" method in the movie model? I thought it was more restful for it to be in the actor model?

File: spec/features/actors/show_spec.rb [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+		it 'I see all of the movies they were in' do
+
+			visit "/actors/#{@lady_gaga.id}"
+
+			expect(current_path).to eq("/actors/#{@lady_gaga.id}")
+
+			expect(page).to have_content(@trolls.name)
+			expect(page).to have_content(@snow.name)
+			expect(page).to have_content(@star.name)
```
[the remark this answers is not in the document]
Comment (sasloan) [reply spec/features/actors/show_spec.rb]: Oh I see so you were looking for an edge case test then. okay I can see that. I can add that easily enough.

File: spec/features/movies/show_spec.rb [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+			@shrek.actors << @mel
+			@shrek.actors << @finn
+
+			@trolls.actors << @lady_gaga
+
+			@turbo.actors << @wyatt
+			@turbo.actors << @mel
+			@turbo.actors << @finn
+			@turbo.actors << @june
```
[the remark this answers is not in the document]
Comment (sasloan) [reply spec/features/movies/show_spec.rb]: okay, I did over do my each block and probably cost me the time I needed to finish user story 4. I will make sure not to repeat that error in the future.

File: spec/features/movies/show_spec.rb [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+		it 'I see a list of all the actors from youngest to oldest' do
+
+			visit "/movies/#{@turbo.id}"
+
+			expect(current_path).to eq("/movies/#{@turbo.id}")
+
+			expect(page).to have_content(@wyatt.name)
+			expect(page).to have_content(@finn.name)
+			expect(page).to have_content(@mel.name)
+			expect(page).to have_content(@june.name)
```
[the remark this answers is not in the document]
Comment (sasloan) [reply spec/features/movies/show_spec.rb]: I was a little confused about that as well. I decided not to dwell on it in favor of getting more done. Could you give me an example of how we would test for order in the features?

File: spec/models/actor_spec.rb [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+			@wyatt = Actor.create!(name: "Wyatt Oleff", age: 16)
+		end
+
+		it '.list_alphabetically' do
+			expect(Actor.list_alphabetically).to eq([@brad.name, @finn.name, @june.name, @lady_gaga.name, @mel.name, @wyatt.name])
+		end
+
+		it '.average_actor_age' do
+			expect(Actor.average_actor_age).to eq(48.5)
+		end
```
[the remark this answers is not in the document]
Comment (sasloan) [reply spec/models/actor_spec.rb]: Thank you

File: app/views/movies/show.html.erb (hunk 3 above)
[the remark this answers is not in the document]
Comment (mikedao) [reply app/views/movies/show.html.erb]: The latter. Having the method in the actor model has nothing to do with it being RESTful or not.

File: spec/features/movies/show_spec.rb (hunk 6 above)
[the remark this answers is not in the document]
Comment (mikedao) [reply spec/features/movies/show_spec.rb]: I would use a combination of ruby and CSS in order to create unique CSS ids that I could then test.
